codegen/smithy-dafny-codegen-modules/smithy-python/codegen/smithy-python-codegen/src/main/java/software/amazon/smithy/python/codegen/ConfigGenerator.java

```diff
@@ -224,25 +224,29 @@ private void writeInterceptorsType(PythonWriter writer) {
         writer.getImportContainer().addImport("smithy_python.interfaces.interceptor", "Interceptor", "Interceptor");
 
         writer.writeInline("_ServiceInterceptor = Union[");
-        var iter = operationShapes.iterator();
-        while (iter.hasNext()) {
-            var operation = iter.next();
-            var input = symbolProvider.toSymbol(context.model().expectShape(operation.getInputShape()));
-            var output = symbolProvider.toSymbol(context.model().expectShape(operation.getOutputShape()));
-
-            // TODO: pull the transport request/response types off of the application protocol
-            writer.addStdlibImport("typing", "Any");
-            writer.writeInline("Interceptor[$T, $T, Any, Any]", input, output);
-            if (iter.hasNext()) {
-                writer.writeInline(", ");
-            } else {
-                writer.writeInline("]");
+        if (operationShapes.isEmpty()) {
+            writer.writeInline("]");
```

Comment:
Much more of a nit now, but seems like you could just move the `writer.writeInline("]");` to after the loop if you wanted.